### DISCUSSION TITLE
Way to ascension softlock if choosing to think about it during waytoascension2

### DISCUSSION THREAD

**Discussion by Vykn** [FakeFishGames/Barotrauma#13699] [Bug reports]
### What happened?

Yet another husk questline bug :barodev:
Probably went unnoticed because nobody would choose this dialogue option

When doing waytoascension2, if you decide not to join the cult after handing over the eggs, there will be no future opportunity to do so, and so the questline becomes impossible to complete

waytoascension2 can only start if the metadata is at 2, but after the eggs are handed over and before you join the cult, the data is set to 3

There should probably be an extra event that can trigger if the waytoascension state is at 3 to offer the crew to join the cult, so that if the player undocks and goes to another station, they may be offered again.

### Reproduction steps

1. start a campaign
2. setcampaign metadata waytoascension 2
3. triggerevent waytoascension2
4. give some eggs, then decline to join the cult
5. undock
6. observe that progressing the cultist questline is now impossible

### Version

v1.3.0.0/v1.3.0.1

**Comment by Regalis11** [comment]
Thank you for the report! Created a ticket about this: https://github.com/FakeFishGames/Barotrauma/issues/13712.

It doesn't seem like the state is the issue though: there is a second part in the event which handles the 3rd state. Looks like the problem is the NPCFollowAction that tries to target the triggerer of the event when it hasn't been set yet, causing the event to get stuck.

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/b628b6396b3f2a7019bb0b8bba620dbf89dd8108
